x-pack/test/functional/services/ml/expanded_job_details.ts

```diff
@@ -0,0 +1,124 @@
+
+        await testSubjects.setValue('mlAnnotationsFlyoutTextInput', newAnnotation, {
+          clearWithKeyboard: true,
+        });
+        await testSubjects.click('annotationFlyoutUpdateOrCreateButton');
+      });
+
+      await jobTable.withDetailsOpen(jobId, async () => {
+        await jobTable.openAnnotationsTab(jobId);
+        if (await testSubjects.exists('clearSearchButton'))
+          await testSubjects.click('clearSearchButton');
+        const visibleText = await testSubjects.getVisibleText(
+          jobTable.detailsSelector(jobId, 'mlAnnotationsColumnAnnotation')
+        );
+        expect(visibleText).to.be(newAnnotation);
+      });
+    },
+
+    async assertDataFeedFlyout(jobId: string): Promise<void> {
+      await jobTable.withDetailsOpen(jobId, async () => {
+        await jobTable.openAnnotationsTab(jobId);
+        if (await testSubjects.exists('clearSearchButton'))
+          await testSubjects.click('clearSearchButton');
+        await testSubjects.click(jobTable.detailsSelector(jobId, 'euiCollapsedItemActionsButton'));
+        await testSubjects.click('mlAnnotationsActionViewDatafeed');
+        await testSubjects.existOrFail('mlAnnotationsViewDatafeedFlyoutChart');
+        const visibleText = await testSubjects.getVisibleText(
+          'mlAnnotationsViewDatafeedFlyoutTitle'
+        );
+        expect(visibleText).to.be(`Datafeed chart for ${jobId}`);
+        await testSubjects.click('euiFlyoutCloseButton');
+      });
+    },
+
+    async openModelSnapshotTab(jobId: string) {
+      await retry.tryForTime(3_000, async () => {
+        await jobTable.ensureDetailsOpen(jobId);
+        await testSubjects.click(jobTable.detailsSelector(jobId, 'mlJobListTab-modelSnapshots'));
+      });
+    },
+
+    async assertModelSnapshotManagement(jobId: string): Promise<void> {
+      await jobTable.withDetailsOpen(jobId, async () => {
+        await this.openModelSnapshotTab(jobId);
+        await testSubjects.existOrFail('mlADModelSnapShotRevertButton');
+        await testSubjects.existOrFail('mlADModelSnapShotsEditButton');
+      });
+    },
+
+    async selectAllJobs(): Promise<void> {
+      await testSubjects.click('checkboxSelectAll');
+    },
+
+    async assertJobListMultiSelectionText(expectedMsg: string): Promise<void> {
+      const visibleText = await testSubjects.getVisibleText('~mlADJobListMultiSelectActionsArea');
+      expect(visibleText).to.be(expectedMsg);
+    },
+
+    async assertColumnState(buttonText: string, columnJobState: string): Promise<void> {
+      await find.clickByButtonText(buttonText);
+      const visibleText = await testSubjects.getVisibleText('mlJobListColumnJobState');
+      expect(visibleText).to.be(columnJobState);
+    },
+
+    async assertColumnId(expectedId: string): Promise<void> {
+      const visibleText = await testSubjects.getVisibleText('mlJobListColumnId');
+      expect(visibleText).to.be(expectedId);
+    },
```

Comment:
I'm not exactly sure what this assertion is supposed to do. It looks very similar to `assertJobRowJobId` that already have in the `job_table` service.

Reply:
I think how you're using this is to make sure that after applying a filter to the job table, only one row with a certain job id exists in the table. Did I understand this correctly?

Reply:
Yes

Reply:
In that case, I'd suggest to make this service method a bit more flexible and re-usable by making it assert the currently existing job ids in the table, e.g. something like `assertJobsInTable(expectedJobIds: string[])` and move that method to the job table service (as it's not related to the expanded row.